src/player/process-death.cpp

```diff
@@ -42,172 +42,149 @@
  * Centers a string within a GRAVE_LINE_WIDTH character string		-JWT-
  * @details
  */
-static void center_string(char *buf, concptr str)
+static std::string center_string(const std::string &str)
 {
-    int i = strlen(str);
-    int j = GRAVE_LINE_WIDTH / 2 - i / 2;
-    (void)sprintf(buf, "%*s%s%*s", j, "", str, GRAVE_LINE_WIDTH - i - j, "");
+    int j = GRAVE_LINE_WIDTH / 2 - str.length() / 2;
+    return std::string().append(j, ' ').append(str).append(GRAVE_LINE_WIDTH - str.length() - j, ' ');
```

Comment:
Instead of `std::string().append(j, ' ')`, the constructor `std::string(j, ' ')` can be used.

Reply:
The latest forced push tightens that to get rid of the call to the default constructor.